cluster/src/main/java/org/apache/iotdb/cluster/log/manage/RaftLogManager.java

```diff
@@ -582,14 +585,35 @@ public void commitTo(long newCommitIndex) throws LogExecutionException {
           .clear();
     }
     try {
-      int currentSize = committedEntryManager.getTotalSize();
-      int deltaSize = entries.size();
-      if (currentSize + deltaSize > maxNumOfLogsInMem) {
-        int sizeToReserveForNew = maxNumOfLogsInMem - deltaSize;
+      boolean needToCompactLog = false;
+      int numToReserveForNew = minNumOfLogsInMem;
+      if (committedEntryManager.getTotalSize() + entries.size() > maxNumOfLogsInMem) {
+        needToCompactLog = true;
+        numToReserveForNew = maxNumOfLogsInMem - entries.size();
+      }
+
+      long newEntryMemSize = 0;
+      for (Log entry : entries) {
+        if (entry.getByteSize() == 0) {
+          logger.warn("{} should not go here, must be send to the follower, so the log has been serialized", entry);
+          newEntryMemSize += entry.serialize().array().length;
```

Comment:
It always goes here when there is only one replica.
By the way, it is wasteful to serialize the log here when there is only one replica. Maybe we can use a rough estimation or leave it to the restriction of the number of entries.

Reply:
1. I serialize the log before entering commitTo when there is only one replica. So it's not right to here. 
2. Yes, it's wasteful to serialize the log when there is only one replica.  Maybe we could pass it.